packages/wonder-blocks-button/src/components/button-core.tsx

```diff
@@ -429,34 +452,23 @@ export const _generateStyles = (
                 borderWidth: theme.border.width.secondary,
                 paddingLeft: padding,
                 paddingRight: padding,
+                // TODO(WB-1844): Change this when we get final designs for
+                // hover.
+                [":hover:not([aria-disabled=true])" as any]: focusStyling,
+                [":focus-visible:not([aria-disabled=true])" as any]:
+                    focusStyling,
+                [":active:not([aria-disabled=true])" as any]:
```

Comment:
Is there something we can do to prevent having `as any`? 

I was trying to see if I ran into similar things when working on the SelectOpener and it looks like we don't type cast it explicitly because we set `newStyles` in SelectOpener to `Record<String, any>` instead of `Record<String, CSSProperties>` 😅 

https://github.com/Khan/wonder-blocks/blob/c58e3fa49e7eba8c4235d6e3c805a5226e1b42d2/packages/wonder-blocks-dropdown/src/components/select-opener.tsx#L294-L304

https://github.com/Khan/wonder-blocks/blob/c58e3fa49e7eba8c4235d6e3c805a5226e1b42d2/packages/wonder-blocks-dropdown/src/components/select-opener.tsx#L258

Reply:
ugh yeah that's very annoying tbh. This is one of the things I'd hope Aphrodite would have better support for. We could add types, but it's hard to calculate all the possible combinations/variants for these selectors.

I guess this is a good/solid point for switching to CSS selectors (aka css-modules). Well, you know that I worry about opening the door to specificity, but that's something that we'll have to evaluate when we make the definitive decision to switch our styling approach.

Reply:
Maybe we could do something related to typing 🤔 
https://www.typescriptlang.org/play/?#code/C4TwDgpgBACgzhArgEwPYGEA2BDOC5QC8UAUFFAD5QBEAXABaoBuEATtWZTbQGaoDGiOB3JU6fQXAC0TAJZxZAI0wQRXOtn7BZLagG5OJUJCgAxRADsts1BeyZ4SNFlz4inMQ1xrP8n9wtUYH0SI3BoAEFWWWwI4GBoxURgCAJCDygAcgBtbGjsKWR5bGUIZABdTIycvJipWQsme1kKqtEs3PypVggAR0RZHtaDclDjSPjE5IgAZQgVLVRWIigAAwASAG9HFAwcPFSAXy3zK21bex3nffxDgAotqJi4hKVpuEOASlWDEn5bODAKBNTAtAASzDYcwWwCWtCgLymKWhEEWy2IdEYLFYtECwDunTqRTgJRUFU+IRIAHoqVAALbyBQWADmUB4lmsF0wUDACF2Un4NygdwgADpmaKoAxIaxPn8AUCGiCWqYBEIUWj4Yi3sj5qjYeiAkECbUCkrmuT9EA

Reply:
Oooh cool! That TS example looks promising! Looks like the ordering would matter. For example, `:not([aria-disabled]):hover` wouldn't meet the type requirements like `:hover:not([aria-disabled])`